llm/cpp/llm.cpp

```diff
@@ -50,37 +33,28 @@ int main(int argc, char* argv[]) try {
     ov::InferRequest detokenizer = core.compile_model(argv[3], "CPU").create_infer_request();
     std::shared_ptr<ov::Model> model = core.read_model(argv[1]);
     constexpr size_t BATCH_SIZE = 1;
-    std::map<std::string, ov::PartialShape> shapes = {
-        {"input_ids", ov::PartialShape{
-            BATCH_SIZE, {1, std::numeric_limits<ov::Dimension::value_type>::max()}
+    std::map<size_t, ov::PartialShape> shapes = {
+        {0, ov::PartialShape{
+            BATCH_SIZE, -1
         }},
-        {"attention_mask", ov::PartialShape{
-            BATCH_SIZE, {1, std::numeric_limits<ov::Dimension::value_type>::max()}
+        {1, ov::PartialShape{
+            BATCH_SIZE, -1
         }}
     };
-    for (const ov::Output<ov::Node>& input : model->inputs()) {
-        for (const std::string& name : input.get_names()) {
-            if (name.rfind("past_key_values", 0) == 0) {
-                ov::PartialShape shape = input.get_partial_shape();
-                shape[0] = BATCH_SIZE;
-                shapes.emplace(name, shape);
-                break;
-            }
-        }
+    std::vector<ov::Output<ov::Node>> inputs = model->inputs();
+    for (size_t idx = 2; idx < inputs.size(); ++idx) {
+        ov::PartialShape shape = inputs.at(idx).get_partial_shape();
+        shape[0] = BATCH_SIZE;
```

Comment:
Set batch for all inputs in one loop, so start with `idx = 0` and remove initializer for `shapes` above. Dimension 0 is affected only in all inputs, no need to handle `input_ids` and `attention_mask` separately.

To better utilize OV capabilities, we can built-in layouts to the original model in optimum-intel export, and use `set_batch` function here here which will affect all inputs automatically. https://docs.openvino.ai/2023.1/openvino_docs_OV_UG_ShapeInference.html#the-set-batch-method.

Reply:
I need to set channel dim to -1 because one of the models is exported by the notebook has static value here. So I can't remove shapes initializer

Reply:
Can we treat it a model issue? In case if we always expect incorrectly set shapes in loaded models, we always need to set all the dimensions. Does it really make sense if we control both parties?

Reply:
I don't know. @eaidova, can I treat this as the notebook issue and ignore it in my implementation?